src/libraries/System.Reflection/tests/ILLink.Descriptor.xml

```diff
@@ -0,0 +1,7 @@
+<linker>
+  <assembly fullname="System.Private.CoreLib">
+      <type fullname="System.Collections.ArrayList" />
```

Comment:
Why is this necessary? It shouldn't be.

Reply:
I was seeing this as:
```
02:19:20] fail: [FAIL] System.Reflection.Tests.TypeInfoTests.IsAssignable(type: typeof(System.Collections.IList), c: typeof(System.Collections.ArrayList), expected: True)
[02:19:20] info: Assert.Equal() Failure
[02:19:20] info: Expected: True
[02:19:20] info: Actual:   False
[02:19:20] info:    at System.Reflection.Tests.TypeInfoTests.IsAssignable(Type type, Type c, Boolean expected)
[02:19:20] info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```
https://github.com/radical/runtime/commit/3cdf08a0ece59578afabc3ffb661de8f7a07d122

This PR is an attempt to remove the trimming errors, to enable AOT tests to be built. I didn't delve into why the linker wasn't being able to pick up the usage. We could look into that, when I make a clean PR based on `main`.